src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -31,7 +32,13 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
             if (objectResult != null)
             {
-                SetContentTypes(objectResult.ContentTypes);
+                // Check if FormatFilter has already set the content type
+                // If it has, dont override it
+                var formatFilter = context.Filters.LastOrDefault(f => f is IFormatFilter) as IFormatFilter;
```

Comment:
You can simplify this to `context.Filters.OfType<IFormatFilter>().LastOrDefault();`